agent/lib/kontena/service_pods/creator.rb

```diff
@@ -78,21 +80,44 @@ def perform
         service_container
       end
 
-      # @param [Docker::Container] service_container
-      # @param [String] type
       # @return [Boolean]
-      def run_hooks(service_container, type)
-        service_pod.hooks.each do |hook|
-          if hook['type'] == type
-            info "running #{type} hook: #{hook['cmd']}"
-            command = ['/bin/sh', '-c', hook['cmd']]
-            log_hook_output(service_container.id, ["running #{type} hook: #{hook['cmd']}"], 'stdout')
-            stdout, stderr, exit_code = service_container.exec(command)
-            log_hook_output(service_container.id, stdout, 'stdout')
-            log_hook_output(service_container.id, stderr, 'stderr')
-            if exit_code != 0
-              raise "Failed to execute hook: #{hook['cmd']}"
+      def run_pre_start_hooks
+        hooks_for('pre_start').each do |hook|
+          service_config = config_container(service_pod.dup)
+          service_config['Cmd'] = ['/bin/sh', '-c', hook['cmd']]
+          service_container = nil
+          begin
+            service_container = create_container(service_config)
+            info "running pre_start hook: #{hook['cmd']}"
+            service_container.tap(&:start).attach { |stream, chunk|
+              log_hook_output(service_container.id, [chunk], stream)
```

Comment:
In order to ensure that the pre-start hook logs actually get picked up, it might be better to fix this the other way: create the hook containers with `io.kontena.container.skip_logs=1` and use the `attach do log_hook_output` instead?